app/models/test_result.rb

```diff
@@ -5,4 +5,17 @@ def readonly?
     true
   end
 
+  def self.slowest
+    select('name, AVG(runtime) AS average_runtime').
+      where(status: 'passed').
+      group(:name).
+      order('average_runtime DESC')
+  end
+
+  def self.by_failure_count
+    select("name, COUNT(CASE WHEN status = 'failed' THEN 1 ELSE NULL END) AS failure_count").
+      group('name').
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.